collection/listpack/listpack.go

```diff
@@ -0,0 +1,138 @@
+package listpack
+
+/* Список имеет указатель на "головной" элемент
+ * каждый элемент имеет указатель на список и, стало быть, указатель на головной элемент.
+ * Если prev или next элемента указывает на "головной элемент", то это значит - конец/начало списка
+ */
+
+type list struct {
+	head *element
+	len int
+}
+
+type element struct {
```

Comment:
Please, create struct in a separate file.
1 struct (and methods for the struct) 1 file